Gateway/Http/Client/TransactionAuthorization.php

```diff
@@ -70,7 +70,12 @@ public function placeRequest(\Magento\Payment\Gateway\Http\TransferInterface $tr
     {
         $request = $transferObject->getBody();
 
-        $requestOptions['idempotencyKey'] = $request['reference'];
+        $requestOptions = [];
+
+        if (!empty($request['idempotencyKeyRequired']) && $request['idempotencyKeyRequired'] === true) {
+            $requestOptions['idempotencyKey'] = $request['reference'];
+            unset($requestOptions['idempotencyKey']);
```

Comment:
why do we first set it and then unset ? probably you want to unset idempotencyKeyRequired right ?  I don't think it is nice to add it into the request pure to know if it is needed is there no elegance way to do this ?

Reply:
Yes indeed, by mistake I unset the wrong one. I will change that.
I think there is a bit more elegant way. I can use the TransferInterface's header property instead of the body to pass it since at the end it is a header setting.

Reply:
Made the changes